docs/features/wired-mesh.rst

```diff
@@ -56,32 +56,63 @@ the ``mesh`` role to the ``interfaces.*.default_roles`` options in site.conf.
 Commandline
 ===========
 
+Since version 2022.1, the wired network configuration is rebuilt from ``/etc/config/gluon`` on each ``gluon-reconfigure``, and therefore overwritten at least on every firmwareupgrade.
+Every interface has a list of roles assigned to it.
+There are currently three roles available: ``client``, ``mesh`` and ``uplink``.
+
+Be careful: When the client role is assigned to an interface at the same time as other roles (like 'client', 'mesh' in the Mesh-on-LAN example below), the other roles take precedence (enabling mesh, but not client in said example).
+
+The setup/config-mode interface is every interface with the role ``client``, which makes removing it from interfaces not only unnecessary, but generally unrecommended.
+
+In order to make persistent changes to the routers configuration it's necessary to:
+
+* change the sections in ``/etc/config/gluon`` e.g. using uci (see examples below)
+* call ``gluon-reconfigure`` to regenerate ``/etc/config/network`` afterwards
+* apply the networking changes, either through executing ``service network restart`` or by performing a ``reboot`` of the device
+
 Enable Mesh-on-WAN::
 
-  uci set network.mesh_wan.disabled=0
-  uci commit network
+  uci add_list gluon.iface_wan.role='mesh'
+  uci commit gluon
 
 Disable Mesh-on-WAN::
 
-  uci set network.mesh_wan.disabled=1
-  uci commit network
+  uci del_list gluon.iface_wan.role='mesh'
+  uci commit gluon
 
 Enable Mesh-on-LAN::
 
-  uci set network.mesh_lan.disabled=0
-  for ifname in $(cat /lib/gluon/core/sysconfig/lan_ifname); do
-    uci del_list network.client.ifname=$ifname
-  done
-  uci commit network
+  uci add_list gluon.iface_lan.role='mesh'
+  uci commit gluon
 
 Disable Mesh-on-LAN::
 
-  uci set network.mesh_lan.disabled=1
-  for ifname in $(cat /lib/gluon/core/sysconfig/lan_ifname); do
-    uci add_list network.client.ifname=$ifname
-  done
-  uci commit network
+  uci del_list gluon.iface_lan.role='mesh'
+  uci commit gluon
+
```

Comment:
```suggestion

For devices with a single Network Port the commands are:

```

Reply:
I added them lower case.

Reply:
4a2c4dbc675f37c9fcabe5e30fd9e75286a9ee98

Reply:
I'm not sure I like that too much.
But the idea very. Maybe I'd reword it to get rid of the colon, as it doubles in the next line.

`For devices with a single network port, instead of `iface_lan` and `iface_wan` configuration is done on `iface_single` as it's more specific.

Reply:
> For single-port-devices, instead of `iface_lan` and `iface_wan` configuration is done in `iface_single`.

Reply:
f988863c4403d6f8d809cbf9cfdd86f70c0db904